include/storage/serialization.hpp

```diff
@@ -266,14 +273,17 @@ template <typename K, typename V> void write(io::BufferWriter &writer, const std
     }
 }
 
-inline void read(io::BufferReader &reader, DataLayout &layout) { read(reader, layout.blocks); }
-
-inline void write(io::BufferWriter &writer, const DataLayout &layout)
+inline void read(io::BufferReader &reader, std::unique_ptr<BaseDataLayout> &layout)
 {
-    write(writer, layout.blocks);
-}
-}
+    read(reader, layout->blocks);
 }
+
+inline void write(io::BufferWriter &writer, const std::unique_ptr<BaseDataLayout> &layout)
```

Comment:
Same here: `const unique_ptr&` makes no sense. If you want a read-only ref to a `BaseDataLayout` then pass by `const BaseDataLayout&`. The unique ptr carries ownership semantics but you are misusing it here.